### PR TITLE
(Fix export_test): change reactor functions which return AlreadyExists err

### PR DESCRIPTION
**Pull request by Longchuanzheng**
**What this PR does / why we need it**:

Fix export_test:
1) Chenge reactor functions return nil obj with AlreadyExists err 
2) The parameters for errors.NewAlreadyExists should be an obj name

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
I think there may be an issue with the parameters passed into the NewAlreadyExists method in the export test.
```
return true, secret, errors.NewAlreadyExists(schema.GroupResource{}, "already exists")
```
"already exists" should be an obj name.

As well as, Reactor functions should return nil on the returned value when the returned error is not nil.
We have problems in clone_test too.

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/storage/export/export/export_test.go [hunk 1]
```diff
@@ -974,8 +974,7 @@ var _ = Describe("Export controller", func() {
 			Expect(ok).To(BeTrue())
 			Expect(secret.GetName()).To(Equal(controller.getExportSecretName(testExportPod)))
 			Expect(secret.GetNamespace()).To(Equal(testNamespace))
-			secret.Name = "something"
-			return true, secret, errors.NewAlreadyExists(schema.GroupResource{}, "already exists")
+			return true, nil, errors.NewAlreadyExists(schema.GroupResource{}, secret.Name)
```
Comment (victortoso) [inline pkg/storage/export/export/export_test.go:977]: The change looks correct to me. As in both cases we don't return error if `errors.IsAlreadyExists`, test result does not change.. I guess you found this by code inspection?

**Comment by fossedihelm** [comment]
@Longchuanzheng Thanks for the contribution. Can I ask you to change the commit message, providing a description of what the problem is and how it is solved? Thank you!

**Comment by Longchuanzheng** [comment]
> @Longchuanzheng Thanks for the contribution. Can I ask you to change the commit message, providing a description of what the problem is and how it is solved? Thank you!

Do you think this is okay?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @Longchuanzheng.
Across the project, there are probably multiple occurrences of returning an object instead of `nil` when a fake client simulates a failure.
Fixing it may also require changes to production code (for example #10903).
Please consider focusing on a single unit test at a time in order to limit the scope of the change.

**Comment by Longchuanzheng** [comment]
> Thank you for the PR @Longchuanzheng. Across the project, there are probably multiple occurrences of returning an object instead of `nil` when a fake client simulates a failure. Fixing it may also require changes to production code (for example #10903). Please consider focusing on a single unit test at a time in order to limit the scope of the change.

Okay, I'll separate it.

**Comment by 0xFelix** [comment]
> As well as, Reactor functions should return nil on the returned value when the returned error is not nil.

Can you elaborate? Is this written in the docs or somewhere else?

**Comment by Longchuanzheng** [comment]
> > As well as, Reactor functions should return nil on the returned value when the returned error is not nil.
> 
> Can you elaborate? Is this written in the docs or somewhere else?

In my opinion, in a real client implementation, the returned value will be nil when the returned error is not nil.
We discussed here  https://github.com/kubevirt/kubevirt/pull/10918#issuecomment-1868470109 and this pr https://github.com/kubevirt/kubevirt/pull/10903 can prove it.

By the way, this pr https://github.com/kubevirt/kubevirt/pull/10936 also hopes to fix this in clone test.

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Comment by dhiller** [comment]
/test pull-kubevirt-e2e-k8s-1.29-ipv6-sig-network
/test pull-kubevirt-e2e-kind-sriov 

Not sure why, but the test lanes are somehow in pending state. I've retriggered them manually.

**Review by dhiller — COMMENTED** [review]
/approve

**Pull request merged**